### DISCUSSION TITLE
Relation between minimal example and emqx-full.yaml

### DISCUSSION THREAD

**Discussion by klemen-df** [emqx/emqx-operator#635] [Q&A] [answered]
Hi,

so, in the example [here ](https://github.com/emqx/emqx-operator/blob/main/docs/en_US/getting-started/getting-started.md#deploy-emqx-5)([click!](https://github.com/emqx/emqx-operator/blob/main/docs/en_US/getting-started/getting-started.md#deploy-emqx-5)) we can see (I think) the minimum needed that the emqx operator deploys emqx.

Then we have [emqx-full.yaml (click!)](https://github.com/emqx/emqx-operator/blob/main/config/samples/emqx/v2alpha1/emqx-full.yaml) where :

- there are many more fields
- some are commented

So, let's take the minimal example ; even if we don't define liveness probe, the probe is there. On the other hand, if we don't define 'runAsUser' then that field isn't there.

What's the relation there?

**Comment by Rory-Z** [comment, marked as answer]
Hi, @KlemenDanfoss In most cases, you only need minimum, the the [emqx-full.yaml ](https://github.com/emqx/emqx-operator/blob/main/config/samples/emqx/v2alpha1/emqx-full.yaml) is just an example, it more friendly than [reference document](https://github.com/emqx/emqx-operator/blob/main/docs/en_US/reference/v2alpha1-reference.md)

**Reply by klemen-df** [reply]
Hm,

@Rory-Z , how come then, that probes (for example) get added even if I didn't add them in my manifest (but they are in emqx-full.yaml)?

**Reply by Rory-Z** [reply]
Sorry for delay, EMQX Operator will add default value via [mutatingwebhookconfigurations.admissionregistration.k8s.io](https://kubernetes.io/docs/reference/kubernetes-api/extend-resources/mutating-webhook-configuration-v1/), sorry we didn't have document about this, maybe you can check the code: https://github.com/emqx/emqx-operator/blob/main/apis/apps/v2alpha1/emqx_webhook.go#L49

**Reply by klemen-df** [reply]
Hm, yes, quite tricky. Thank you for the answer.
